loky/backend/utils.py

```diff
@@ -84,3 +94,221 @@ def _recursive_terminate(pid):
+
+    def mkl_get_max_threads(self):
+        if self.lib_mkl is None:
+            raise NotImplementedError("Could not find MKL libray")
+        return self.lib_mkl.MKL_Get_Max_Threads()
+
+    def _load_lib(self, module_name, lib_info):
+        lib_name = find_library(module_name)
+        if lib_name is not None:
+            return ctypes.CDLL(lib_name, use_errno=True)
+        return self._get_lib_from_dynamic_libs(module_name)
+
+    # def _get_lib_from_numpy(self, module_name, lib_info):
+    #     from glob import glob
+    #     try:
+    #         from numpy import __config__
+    #         lib_info = getattr(__config__, lib_info, {})
+
+    #         if lib_info:
+    #             LIBRARY_GLOB = "{{}}/*{}*".format(module_name)
+    #             for folder in lib_info['library_dirs']:
+    #                 lib_names = glob(LIBRARY_GLOB.format(folder))
+    #                 if len(lib_names) > 0:
+    #                     return ctypes.CDLL(lib_names[0], use_errno=True)
+    #     except ImportError:
+    #         pass
+
+    #     if sys.platform != "win32":
+    #         return self._get_lib_from_dynamic_libs(module_name)
+
+    def _get_lib_from_dynamic_libs(self, module_name):
+
+        global _MODULE_NAME
+        _MODULE_NAME = None
+
+        libc_name = find_library("c")
+        if libc_name is None:
+            return
+        libc = ctypes.CDLL(libc_name)
+        if not hasattr(libc, "dl_iterate_phdr"):
+            return
+
+        signature_cfunc = ctypes.CFUNCTYPE(
+            ctypes.c_int, ctypes.c_void_p, ctypes.c_size_t, ctypes.c_char_p)
+        c_check_module = signature_cfunc(check_module)
+
+        data = ctypes.c_char_p(module_name.encode('utf-8'))
+        res = libc.dl_iterate_phdr(c_check_module, data)
+        if res == 1:
+            return ctypes.CDLL(_MODULE_NAME)
+
+
+_clibs_wrapper = None
+
+
+def _get_wrapper():
+    global _clibs_wrapper
+    if _clibs_wrapper is None:
+        _clibs_wrapper = _CLibsWrapper()
+    return _clibs_wrapper
+
+
+def limit_threads_clib(max_threads_per_process):
+    """Limit the number of threads available for openblas, mkl and openmp
+
+    Set the maximal number of thread that can be used for these three libraries
+    to `max_threads_per_process`. This function works on POSIX plateforms and
+    can be used to change this limit dynamically.
+
+    Return a dict dynamic_scaling containing pairs `('clib': boolean)` which
+    are True if `clib` have been found and can be used to scale the maximal
+    number of hreads dynamically.
+    """
+    return _get_wrapper().limit_threads_clibs(max_threads_per_process)
+
+
+def get_thread_limits():
+    """Return thread limt set for openblas, mkl and openmp
+
+    Return a dictionary containing the maximal number of threads that can be
+    used for these three library or None if this library is not available. The
+    key of the dictionary are {'mkl', 'openblas', 'openmp_gnu', 'openmp_intel'}
```

Comment:
We might also want to add support for `openmp_clang` at some point in the future (probably in another PR).